libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/WebSocketConnector.cs

```diff
@@ -0,0 +1,148 @@
+        // TODO: We must document this somewhere, right? Find it and put a reference link here.
+        private const string AuthHeaderName = "authorization";
+        private const string ChannelIdHeaderName = "channelid";
+        private readonly IChannelProvider _channelProvider;
+        private readonly ICredentialProvider _credentialProvider;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebSocketConnector"/> class.
+        /// Constructor for use when establishing a connection with a WebSocket server.
+        /// </summary>
+        /// <param name="credentialProvider">Used for validating channel credential authentication information.</param>
+        /// <param name="channelProvider">Used for validating channel authentication information.</param>
+        /// <param name="logger">Set in order to enable logging.</param>
+        internal WebSocketConnector(ICredentialProvider credentialProvider, IChannelProvider channelProvider = null, ILogger logger = null)
+        {
+            _credentialProvider = credentialProvider;
+            _channelProvider = channelProvider;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Process the initial request to establish a long lived connection via a streaming server.
+        /// </summary>
+        /// <param name="onTurnError"> The function to execute on turn errors.</param>
+        /// <param name="middlewareSet">The set of middleware to perform on each turn.</param>
+        /// <param name="httpRequest">The connection request.</param>
+        /// <param name="httpResponse">The response sent on error or connection termination.</param>
+        /// <param name="cancellationToken">Optional cancellation token.</param>
+        /// <returns>Returns on task completion.</returns>
+        internal async Task ProcessAsync(Func<ITurnContext, Exception, Task> onTurnError, List<Builder.IMiddleware> middlewareSet, HttpRequest httpRequest, HttpResponse httpResponse, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (httpRequest == null)
+            {
+                var ex = new ArgumentNullException(nameof(httpRequest));
+                _logger?.LogError(ex.Message);
+
+                throw ex;
+            }
+
+            if (httpResponse == null)
+            {
+                var ex = new ArgumentNullException(nameof(httpResponse));
+                _logger?.LogError(ex.Message);
+
+                throw ex;
+            }
+
+            if (!httpRequest.HttpContext.WebSockets.IsWebSocketRequest)
+            {
+                httpRequest.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                await httpRequest.HttpContext.Response.WriteAsync("Upgrade to WebSocket is required.").ConfigureAwait(false);
+                _logger?.LogInformation("Invalid request: Request was not a WebSocket handshake.");
+
```

Comment:
Assuming you want to log here (which I'm not clear on), there's not enough info in this logging statement. You would want more details to make it in any way actionable.